test/algorithms/test_warm_start_qaoa.py

```diff
@@ -52,13 +57,19 @@ def test_max_cut(self):
         backend = BasicAer.get_backend("statevector_simulator")
         qaoa = QAOA(quantum_instance=backend, reps=1)
         aggregator = MeanAggregator()
+        # a list of converters without max2min
+        ineq2eq = InequalityToEquality()
+        int2bin = IntegerToBinary()
+        penalize = LinearEqualityToPenalty()
+        converters = [ineq2eq, int2bin, penalize]
```

Comment:
I'm not convinced all of the converters are required to pass to the optimizer in this test. Does the test fail without them?

Reply:
If a converters list with max2min is passed, the problem is converted to a minimization problem before it is given to the presolver. The same happens if no converters list is given, because ``QuadraticProgramToQubo`` is used in that case.

Since in the test case ``test_max_cut`` the presolver is ``GoemansWilliamsonOptimizer``, it fails. 
``GoemansWilliamsonOptimizer`` expects a maximization problem. The ``solve`` method of the presolver is called with the converted_problem ( Line 309). 

I listed the outcomes of the presolver solve method for this test case in a very early comment on the PR:
> The problem is, that the presolver GoemansWilliamsonOptimizer is now getting a minimization qubo. Therefore, opt_result in
> 
> https://github.com/Qiskit/qiskit-optimization/blob/3e21815034c72030eab17f2865895dc0d12076a4/qiskit_optimization/algorithms/warm_start_qaoa_optimizer.py#L294
> is now
> 
> optimal function value: 0.0
> optimal value: [0 0 0 0]
> status: SUCCESS
> 
> Instead of
> optimal function value: 4.0
> optimal value: [0 1 1 0]
> status: SUCCESS
> 
> This leads to a wrong result.
> 
...
> Then, we can alter the UnitTest to use a fixed List of converters(without MaximizeToMinimize) instead of the default list.
> 
> Maybe GoemansWilliamsonOptimizer should give a hint if it is used with a minimization problem?

Reply:
OK, I see the problem now. I believe the warm start optimizer must work without passing additional converters. So, the initial version of the test should stay unchanged. That means we have a few options:

- Do not add max to min in quadratic_program_to_qubo.py. @t-imamichi I guess this is not what we want to achieve, right?
- Modify Goemans-Williamson in a way it converts max to min, may be with an additional warning. This optimizer is built to solve just a very specific type of problems. Sounds messy, requires an additional min to max converter, but still a viable solution and I can take care of this.
- The easiest, tbh, add a check to the warm start if there's a min problem is passed and presolver is Goemans-Williamson, then raise an error saying something like "please provide converters manually". And then we change the test.

@t-imamichi, what do you think? 

@tomtuamnuq I really appreciate your patience and efforts in this PR. It looked simpler initially.

Reply:
I see. My first idea is to include max2min converter in quadratic_to_qubo because all algorithms suppose minimization algorithm. If Goemans-Williamson takes maximization problem, my idea does not work. Let me check the code.

Reply:
I agree with @adekusar-drl's idea not to include max2min converter in qp_to_qubo because Goemans-Williamson is a maximization algorithm. We need to manually append max2min in `self._converters` in some algorithms if necessary.

As for Goemans-Williamson, we need to check whether the sense of the problem is maximization or not in `GoemansWilliamsonOptimizer.get_compatibility_msg`.
Another option is that we make min2max converter and include it as part of GW.

@tomtuamnuq I'm sorry I did not notice Goemans-Williamson is a maximization algorithm. I appreciate your patience and contribution to our code.

Reply:
@t-imamichi Are sure we should remove max2min in qp_to_qubo just because of one optimizer (which is a utility one)? I fine with either solution here anyway.

Changes to Goemans-Williamson is a separate PR, I agree.

Reply:
Hello @adekusar-drl and @t-imamichi , thanks for your feedback on this. I will revert changes in qp_to_qubo. I have checked which algorithms use ``self._prepare_converters`` method: grover, meo, recursive_meo, warm_start_qaoa

grover deals with minimization only. We must make sure max2min is in the list ``self._converters``. 
I would just overwrite the inherited ``_prepare_converters`` method, okay?

warm_start_qaoa(as we have seen) must not use max2min before passing the problem to the presolver.
warm_start_qaoa uses the ``_solve_internal`` method of meo (which assumes a minimization problem with current implementation). 

meo can do maximization problems as well but in that case we need:

https://github.com/Qiskit/qiskit-optimization/blob/93b322f954972dea11a0d4a0957c79d28648e313/qiskit_optimization/algorithms/minimum_eigen_optimizer.py#L222

in method ``_solve_internal`` as before. It should then work.
The same goes for recursive_meo. Since recursive_meo is using other optimizers we should not apply max2min here.

Like discussed before it would be good to do  Goemans-Williamson changes in another PR to keep this PR simple.

Reply:
@adekusar-drl If GW is not supposed to use as an independent optimizer, it might be good to adjust GW as a separate PR. Sorry, I didn't understand the usage of GW. If we take care of GW as a separate PR. We can merge this PR as is. Is it correct?

Reply:
I notice that we maybe don't need any converters in this case. https://github.com/Qiskit/qiskit-optimization/pull/114/files#r636922553

Reply:
@t-imamichi I have tested and yes the empty list works.

Reply:
Yes, it is because maxcut is a QUBO even without any conversion. Let's wait for a response of @adekusar-drl

Reply:
@t-imamichi GW is a helper for the warm start QAOA. It solves a relaxed problem, actually samples possible solutions, and then a customized QAOA circuit is built from the sampled solutions. So, this is a utility optimizer.

As a temporary solution in this PR `converters=[]` is ok, but I'd expect the warm start optimizer to be correctly working even if I don't pass any converters. And this is not the case if this PR is merged as it is now, since if there's no converters, defaults are applied and max is converted to min and GW fails eventually. This is the only thing makes me worry. I'm ok if we stream-line this behavior separately, in another PR.